src/app/space/create/pipelines/pipelines.component.ts

```diff
@@ -116,21 +116,37 @@ export class PipelinesComponent implements OnInit, OnDestroy {
       }));
 
     this.subscriptions.push(
-      this.runtimePipelinesService.current.subscribe((buildConfigs: BuildConfig[]) => {
-        this._allPipelines = buildConfigs;
-        this.applyFilters();
-        this.applySort();
-      }));
+      this.runtimePipelinesService.current.combineLatest(
+        this.pipelinesService.getOpenshiftConsoleUrl().do((url: string) => {
```

Comment:
Does this need to be done? The combineLatest docs say that it does not complete until all of the inner Observables have completed, and any Observables that complete before the rest will essentially hold their last emitted value. I think you already get the behaviour you describe with the way you've written it? Maybe it would be clearer to read if you use the static `Observable.combineLatest()`. I also find it a little confusing that you have a `do` side-effect inside of this `combineLatest` for the console URL - I think the typical usage of a `combineLatest` would have all of the inner Observables emitting their values, and all of the latest values from each stream are then collected into an output array where the indices of this array correspond to the order of the inner Observables. Then you would subscribe on this resulting Observable and update the component state based on the values found within that array.

Reply:
You're right, it'll still work until both observables complete afaict via manual testing. I wasn't able to find documentation though maybe I'm a bit blind this time...

Yeah the `do` side-effect is just a trick to piggy-back the fact that the inner observable will still emit there and I want to do something on each emission specific to that inner observable and not to the combined results.

As for the combined array, that's the default behavior without a function supplied to `combineLatest` to transform it. I decided to supply a transforming function as the goal is to still output the same format as the outer observable, just with modified contents. It can be accomplished both ways, I think this way is actually a bit cleaner.

The use of `combineLatest` via `Observable...` or `instance...` is really a style thing, both do the same thing. I agree `Observable...` is clearer in that it's combining the listed observables but I didn't want to add an import hahah. I can switch it if you'd like. Thoughts?

Reply:
`do` is meant to be used as a utility operator. While this works, it feels wrong.

Reply:
`.do` and `.map` both execute code but `.do` doesn't return or change the value

Reply:
I've extracted the operation into a separate subscription. How does it look?

Reply:
I think it's actually cleaner to reuse the same subscription when possible, especially if it's inside the same function. A developer may come along and wonder why it was put into two subscriptions if it could be done in one. Thoughts?

Reply:
To be clear I don't have a preference for either approach. Given that you both found it quirky, I think it would also make sense to split it up.